### PR TITLE
Handle nested packages better

### PR DESCRIPTION
**Pull request by ejholmes**
Fixes https://github.com/twitchtv/twirp-ruby/issues/13

Given a package name `package code.justin.tv.my_cool_package;`, this will generate a nested set of modules like this:

```ruby
module Code
  module Justin
    module Tv
      module MyCoolPackage
```

Which matches the way protoc-gen-ruby generates the modules.

### CODE REVIEW

File: protoc-gen-twirp_ruby/main.go [hunk 1]
```diff
@@ -217,6 +219,16 @@ func writeResponse(w io.Writer, resp *plugin.CodeGeneratorResponse) {
...
 
+// Modules converts protobuf package name to a list of Ruby module names to
+// represent it.
+func Modules(pkgName string) []string {
+	var parts []string
+	for _, p := range strings.Split(pkgName, ".") {
+		parts = append(parts, CamelCase(p))
+	}
+	return parts
+}
```
Comment (marioizquierdo) [inline protoc-gen-twirp_ruby/main.go]: The name "Modules" is a little bit generic, even for a small file like this. Rename it to something like `PackageToModules` or `GetRubyModules`. The comment can also have an example: `e.g. "my.package" => {"My", "Package"}`.

File: protoc-gen-twirp_ruby/main_test.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.pkgName, func(t *testing.T) {
+			modules := Modules(tt.pkgName)
+			if got, want := modules, tt.modules; !reflect.DeepEqual(got, want) {
+				t.Errorf("expected %v; got %v", want, got)
+			}
+		})
+	}
```
Comment (marioizquierdo) [inline protoc-gen-twirp_ruby/main_test.go]: Thanks for unit testing this! Maybe it is worth adding a case for empty string `""`, package components with 1 letter `"m.v.p"` and invalid packages `"foo.....bar"`.

**Comment by marioizquierdo** [comment]
@ejholmes just merged #15 in master. That changed the `example.HelloWorld` service to `example.hello_world.HelloWorld`, that has 2 package components.

If you pull from master, you can see that it generates an invalid `service_twirp.rb` file, using the wrong Ruby syntax `module Example.helloWorld` instead of nested `module Example\n  module HelloWorld`. You can follow the [example/README.md](https://github.com/twitchtv/twirp-ruby/blob/master/example/README.md) instructions to re-generate the code with the changes you made here, and iterate until it works properly.

**Comment by marioizquierdo** [comment]
FYI, we think this is the last important bug in out plate. After this PR is merged we will call it version `v1.0.0` 😃

**Comment by ejholmes** [comment]
@marioizquierdo alrighty. Made the changes, and it now generates valid Ruby, with appropriate indentation:

```console
$ cd example
$ protoc --proto_path=. ./hello_world/service.proto --ruby_out=. --twirp_ruby_out=.
$ find . -type f -iname '*.rb' | xargs -L1 ruby -c
Syntax OK
Syntax OK
Syntax OK
Syntax OK
```

**Pull request merged**